### PR TITLE
Convenience method to center particles

### PR DESCRIPTION
**Pull request by lyrivera**
## Description

Adds convenience method to the `Box` class to center the particles using periodic boundaries.

## Motivation and Context

Resolves: None

## How Has This Been Tested?

Tested by visual inspection, no tests has been added yet.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [ ] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [ ] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/credits.rst).
- [ ] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: freud/box.pyx [hunk 1]
```diff
@@ -341,6 +341,27 @@ cdef class Box:
...
+        Args:
+            vecs (:math:`\left(3, \right)` or :math:`\left(N, 3\right)` :class:`numpy.ndarray`):
+                Vector(s) to be unwrapped.
+
+        Returns:
+            :math:`\left(3, \right)` or :math:`\left(N, 3\right)` :class:`numpy.ndarray`:
+                Centered vector(s).
+        """  # noqa: E501
+        vecs = np.asarray(vecs)
+        vecs = np.atleast_2d(vecs)
```
Comment (bdice) [inline freud/box.pyx]: Be careful here with 1D vectors. If a single vector is passed in, I don't think this is returning a single vector. It's also not very useful to center one point, so I'd be fine with requiring a 2D array of input data. If you make that change, please alter the docstrings to only allow (N, 3) for inputs and outputs.

**Comment by bdice** [comment]
Some ideas for tests of 2D arrays:

```
box = freud.box.Box.cube(5)
npt.assert_almost_equal(box.center([[0, 0, 0]]), [[0, 0, 0]])
npt.assert_almost_equal(box.center([[1, 1, 1]]), [[0, 0, 0]])
npt.assert_almost_equal(box.center([[1, 1, 1], [2, 2, 2]]), [[-0.5, -0.5, -0.5], [0.5, 0.5, 0.5]])
npt.assert_almost_equal(box.center([[-2, -2, -2], [2, 2, 2]]), [[0.5, 0.5, 0.5], [-0.5, -0.5, -0.5]])
# Something with a triclinic box
```

**Comment by bdice** [comment]
One feature that could be really helpful for people dealing with atoms or polydispersity: add an argument for `masses=None` that defaults to `np.ones(len(positions))` if not provided / `None`.

File: freud/box.pyx (hunk 1 above)
Comment (lyrivera) [reply freud/box.pyx]: Thanks for catching it. I removed the 2D enforcement and let `make_absolute` handle it when returning.

**Comment by lyrivera** [comment]
> One feature that could be really helpful for people dealing with atoms or polydispersity: add an argument for `masses=None` that defaults to `np.ones(len(positions))` if not provided / `None`.

This would be used to do a mass-weighted average, or something along those lines, I assume?

**Comment by vyasr** [comment]
Also, please don't commit the cpp files. We only update those on releases to avoid massive repo bloat, so users working off of master do need to build with Cython.

**Comment by bdice** [comment]
@lyrivera @vyasr After some consideration, I would prefer to implement this in C++, because other parts of freud's C++ code like ClusterProperties currently solve the same problem (though in a less-trustworthy way). I have a draft implementation and will make a PR. Apologies for the inconvenience.

**Comment by lyrivera** [comment]
> @lyrivera @vyasr After some consideration, I would prefer to implement this in C++, because other parts of freud's C++ code like ClusterProperties currently solve the same problem (though in a less-trustworthy way). I have a draft implementation and will make a PR. Apologies for the inconvenience.

Sounds good. I'll close this.

**Pull request closed without merging**